### PR TITLE
Update Steward process

### PR DESCRIPTION
**Pull request by logankilpatrick**
Hey @JuliaLang/stewards this PR adds context as to how someone can become a Steward and the like. Will add a nomination form and get you all access if you think this wording looks good. 

CC @jiahao

### CODE REVIEW

File: community/stewards.md [hunk 1]
```diff
@@ -44,3 +44,13 @@ Note that:
...
 @@
+
+## Becoming a Steward
+
+The Julia Stewards were originally chosen for their long term commitment and quality contributions to the Julia community. Stewards currently serve at will and do not have a pre-defined appointment term. If a steward decides to give up their role and move on, the other stewards may seek out an individual whom they know has been a long term contributor to the Julia community. This is a high trust role given the implications mentioned above, which is generally why there is a need for folks to be trusted in the community before they become a steward. 
+
+In additional to the stewards seeking out folks to add to the committee, you can also nominate someone (including yourself) to join the committee, even if there is not a vacancy. Nominations are accepted on a rolling basis and will be reviewed if/when the current stewards determine they need to bring on additional support. Nominations are reviewed on the axis of long term commitment and contributions to the Julia community (which need not be technical in nature).
```
[the remark this answers is not in the document]
Comment (logankilpatrick) [reply community/stewards.md]: TY!

**Review by timholy — APPROVED** [review]
Approve pending addition of the nomination form

**Comment by ViralBShah** [comment]
A follow-on discussion from stewards - the form needs to also include 3 references. @logankilpatrick can you add those fields? Also, when the form is filled out, who will receive it?

**Comment by logankilpatrick** [comment]
I gave access to all the stewards except @nalimilan (please send me a google address if you have one). Will add the references section

**Comment by logankilpatrick** [comment]
Added references request

**Pull request merged**